yb-voyager/cmd/importData.go

```diff
@@ -382,7 +382,9 @@ func generateSmallerSplits(taskQueue chan *SplitFileImportTask) {
 			log.Infof("clearing the generated splits for table %q matching %q pattern", table, filePattern)
 			utils.ClearMatchingFiles(filePattern)
 		}
-		importTables = allTables //since all tables needs to imported now
+
+		//Removing intersection of doneTables and ExcludeTableList from allTables list, just for startClean
+		importTables = removeExcludeTables(allTables) //since all tables needs to imported now
```

Comment:
why truncate and clear splits for the tables which are not included?
Instead, just apply this filter beforehand on `allTables` and `importTables` lists

Reply:
This specific case applies to when `--start-clean` has been used. The filter has already been applied to `importTables` list by this point. If we wish to truly "start clean", we should empty the all the splits right? That's why I filtered the list after truncating and clearing splits for all tables

Reply:
That is what the user expectation will be.
Here if i am not wrong, you are applying filter after the startClean process is done, in that case you are going to clear already done tables also.

Reply:
What if you just apply the filter at once after line 370 for the `importTables` and `allTables` lists. Once those tables are out/removed there is no need to apply filter again at any place.

Reply:
I am onboard with this idea, and was actually spending a while trying to understand the "expected" behaviour of start clean- what we can do for now is file a separate ticket introducing the separate mode of start clean as we talked about during today's stand-up (one mode is a full wipe, the other mode is only wipe the tables mentioned through the table list/not the ones mentioned in exclude table list). It will help define behaviour for if/when resumability is introduced in export data as well

Reply:
Sure, goahead file a ticket for export data. There is a already existing ticket for export data resumability, link/tag that ticket in the new ticket.

Reply:
> What if you just apply the filter at once after line 370 for the importTables and allTables lists. Once those tables are out/removed there is no need to apply filter again at any place.

I feel, this approach will simplify the code a lot for developers.

Reply:
It may seem simpler, but it will not provide a consistent meaning of what "start clean" means in our code, as of today

This behaviour should then also be reflected on the export side, where `--start-clean` and `--table-list` together should only clean the tables mentioned in the table list, keeping the data for already exported (but not mentioned) tables' data intact